lib/widgets/menu.dart

```diff
@@ -27,21 +27,82 @@ extension ContextMenuPortalEntrySender on BuildContext {
   void closeMenu() => read<_OffsetCubit>().emit(null);
 }
 
+class PopupMenuPageButton<T> extends HookConsumerWidget {
+  const PopupMenuPageButton({
+    super.key,
+    required this.itemBuilder,
+    this.onSelected,
+    this.child,
+    this.icon,
+  });
+
+  final PopupMenuItemBuilder<T> itemBuilder;
+  final PopupMenuItemSelected<T>? onSelected;
+  final Widget? child;
+  final Widget? icon;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) => PopupMenuButton(
+        color: context.theme.popUp,
+        padding: EdgeInsets.zero,
+        constraints: const BoxConstraints(minWidth: 160),
+        shape: const RoundedRectangleBorder(
+            borderRadius: BorderRadius.all(Radius.circular(11))),
+        itemBuilder: itemBuilder,
+        onSelected: onSelected,
+        icon: icon,
+        child: child,
+      );
+}
+
+PopupMenuItem<T> createPopupMenuItem<T>({
```

Comment:
should create A new Widget class which inherits from `PopupMenuItem`, for example: `CheckedPopupMenuItem`.

Using global methods to create Widgets seems a bit awkward.